blocks/recent-articles/recent-articles.js

```diff
@@ -19,74 +16,49 @@ const readNowText = getTextLabel('READ NOW');
-  const limit = extractLimitFromBlock(block) || defaultLimit;
-  const allArticles = await fetchMagazineArticles();
-
-  const allArticlesWithImage = removeArticlesWithNoImage(allArticles);
-  const sortedArticles = sortArticlesByDateField(allArticlesWithImage, 'lastModified');
-  const filteredArticles = clearRepeatedArticles(sortedArticles);
-  const selectedArticles = filteredArticles.slice(0, limit);
-
-  const recentArticlesSection = createElement('div', { classes: `${blockName}-section` });
-  const recentArticlesTitle = createElement('h3', { classes: `${blockName}-section-title` });
-  recentArticlesTitle.innerText = sectionTitle;
-
-  const recentArticleList = createElement('ul', { classes: `${blockName}-list` });
-
-  selectedArticles.forEach((e, idx) => {
+const createList = (articles) => `
+  <ul class="${blockName}-list">
+  ${articles.map((e, idx) => {
+    const firstOrRestClass = (idx === 0) ? 'first' : 'rest';
+    const picture = createOptimizedPicture(e.image, e.title);
+    const pictureTag = picture.outerHTML;
     const linkUrl = new URL(e.path, getOrigin());
-    const firstOrRest = (idx === 0) ? 'first' : 'rest';
 
-    const item = createElement('li', { classes: `${blockName}-${firstOrRest}-item` });
+    const articleCategory = e.category;
+    const categoryWithDash = articleCategory.replaceAll(' ', '-').toLowerCase();
+    const categoryUrl = new URL(`magazine/categories/${categoryWithDash}`, getOrigin());
 
-    const image = createElement('div', { classes: `${blockName}-${firstOrRest}-image` });
-    const picture = createOptimizedPicture(e.image, e.title);
-    const pictureTag = picture.outerHTML;
-    image.innerHTML = `<a href='${linkUrl}'>
-      ${pictureTag}
-    </a>`;
+    return (
+      `<li class="${blockName}-${firstOrRestClass}-item">
+        <div class="${blockName}-${firstOrRestClass}-image">
+          <a href="${linkUrl}">${pictureTag}</a>
+        </div>
+        ${(idx === 0 && e.category) ? `<a class="${blockName}-${firstOrRestClass}-category" href="${categoryUrl}">${e.category}</a>` : ''}
+        <a class="${blockName}-${firstOrRestClass}-title" href="">${e.title}</a>
+        ${(idx === 0 && e.subtitle) ? `<p class="${blockName}-${firstOrRestClass}-subtitle">${e.subtitle}</p>` : ''}
+        ${(idx === 0) ? `<a class="${blockName}-${firstOrRestClass}-link" href="${linkUrl}">${readNowText}</a>` : ''}
+      </li>`
+    );
+  }).join('')}
+  </ul>`;
 
-    const title = createElement('a', {
-      classes: `${blockName}-${firstOrRest}-title`,
-      props: { href: linkUrl },
-    });
-    title.innerText = e.title;
+export default async function decorate(block) {
+  const limit = extractLimitFromBlock(block) || defaultLimit;
 
-    const subtitle = createElement('p', { classes: `${blockName}-${firstOrRest}-subtitle` });
-    subtitle.innerText = e.subtitle;
+  const queryVariables = { limit: limit + 1, sort: 'LAST_MODIFIED_DESC' };
+  const allData = await fetchMagazineData(queryVariables);
```

Comment:
no need to download all of them, use a query to return the amount needed using the limit parameter in the variables sended in the query

Reply:
After yesterday's refinement and discussing how to sort the articles of a block, I’m now considering how we should sort the articles in certain blocks. Previously, we retrieved all articles and sorted them by their publish date, not by their last modified date. Currently, the Search API only supports`LAST_MODIFIED_DESC`, `LAST_MODIFIED_ASC`, and `BEST_MATCH`. To maintain the previous behaviour, we need to retrieve all the articles and handle the sorting manually. For now, I'd suggest leaving it as it is and ask confirmation about how it should be. But the purpose of a "recent-articles" block should be to display the most recently published articles, not the most recently updated ones. In VT, articles from as far back as 2001 could appear first if using last modified.

Reply:
> no need to download all of them, use a query to return the amount needed using the limit parameter in the variables sended in the query

I'm only asking for `limit+1` articles, no all of them

Reply:
and @amglebihan I agree but think that until we have the date metadata field ingested by open search this is the best way we can avoid loading all articles on the FE. So that when that is implemented we can just change the sorting option